src/creature.cpp

```diff
@@ -226,6 +228,36 @@ bool Creature::sees( const Creature &critter ) const
     return sees( critter.pos(), critter.is_player() );
 }
 
+void Creature::remove_carried_creature()
+{
+    carried_creature = cata::nullopt;
+}
+
+void Creature::add_carried_creature( Creature &new_creature )
+{
+    cata::optional<Creature *> p;
+    p = &new_creature;
+    carried_creature = p;
+}
+
+Creature *Creature::get_carried_creature() const
+{
+    if( carried_creature ) {
+        Creature *result;
+        result = *carried_creature;
+        return result;
+    }
+    return nullptr;
+}
+
+bool Creature::has_carried_creature() const
+{
+    if( carried_creature ) {
+        return true;
+    }
+    return false;
+}
```

Comment:
Could be shortened to simple `return carried_creature`.

Reply:
cannot convert ‘const cata::optional<Creature*>’ to ‘bool’ in return  ?